File: projekt-frontend/src/services/PersonService.js
import axios from 'axios';

const PERSON_API_BASE_URL = "http://localhost:8080/api";

class PersonService{
    getPersons(){
        return axios.get(PERSON_API_BASE_URL + "/persons");
    }

    createPerson(person){
        return axios.post(PERSON_API_BASE_URL + "/persons", person);
    }

    getCompanyNames(){
        return axios.get(PERSON_API_BASE_URL + "/company/names");
    }

    getPersonById(personId){
        return axios.get(PERSON_API_BASE_URL + '/persons/' + personId);
    }

    updatePerson(person, personId){
        return axios.put(PERSON_API_BASE_URL + '/persons/' + personId, person);
    }

    deletePerson(personId){
        return axios.delete(PERSON_API_BASE_URL + '/persons/' + personId);
    }



}
export default new PersonService();

File: src/main/java/com/lewandowski/springproject/service/PersonService.java
package com.lewandowski.springproject.service;

import com.lewandowski.springproject.domain.Company;
import com.lewandowski.springproject.domain.Person;

import java.util.List;

public interface PersonService {

    List<Person> getAllPersons();

    Person addPerson (Person person);

    void deletePerson(long id);

    Person updatePerson(long id, Person person);

    Person getPersonById(long id);

    List<Person> getPersonsByCompany(String company);

    List<Company> getCompanyList();

    List<String> getCompanyNames();

}
